galoy-client/src/client/queries.rs

```diff
@@ -0,0 +1,209 @@
+        let user_login = response.user_login;
+        Ok(user_login)
+    }
+}
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "src/client/graphql/schema.graphql",
+    query_path = "src/client/graphql/queries/transactions_list.graphql",
+    response_derives = "Debug, PartialEq, Clone"
+)]
+pub struct StablesatsTransactionsList;
+pub type WalletId = String;
+pub type Timestamp = u64;
+pub type Memo = String;
+pub(crate) type SignedAmount = f64;
+
+pub(crate) type StablesatsTransactions =
+    stablesats_transactions_list::StablesatsTransactionsListMeDefaultAccountTransactions;
+
+impl TryFrom<stablesats_transactions_list::ResponseData> for StablesatsTransactions {
+    type Error = GaloyClientError;
+
+    fn try_from(response: stablesats_transactions_list::ResponseData) -> Result<Self, Self::Error> {
+        let me = response.me;
+        let me = match me {
+            Some(me) => me,
+            None => {
+                return Err(GaloyClientError::GrapqQlApi(
+                    "Empty `me` in response data".to_string(),
+                ))
+            }
+        };
+
+        let default_account = me.default_account;
+        let transactions = default_account.transactions;
+
+        let transactions = match transactions {
+            Some(tx) => tx,
+            None => {
+                return Err(GaloyClientError::GrapqQlApi(
+                    "Empty `transactions` in response data".to_string(),
+                ))
+            }
+        };
+
+        Ok(transactions)
+    }
+}
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "src/client/graphql/schema.graphql",
+    query_path = "src/client/graphql/queries/wallets.graphql",
+    response_derives = "Debug, PartialEq, Clone"
+)]
+pub struct StablesatsWallets;
+pub type StablesatsWalletsWrap = StablesatsWalletsWrapper;
+
+pub struct StablesatsWalletsWrapper {
+    pub btc_wallet: Option<StablesatsWalletsMeDefaultAccountWallets>,
+    pub usd_wallet: Option<StablesatsWalletsMeDefaultAccountWallets>,
+}
+
+impl TryFrom<stablesats_wallets::ResponseData> for StablesatsWalletsWrapper {
+    type Error = GaloyClientError;
+
+    fn try_from(response: stablesats_wallets::ResponseData) -> Result<Self, Self::Error> {
+        let me = response.me;
+        let me = match me {
+            Some(me) => me,
+            None => {
+                return Err(GaloyClientError::GrapqQlApi(
+                    "Empty `me` in response data".to_string(),
+                ))
+            }
+        };
+        let default_account = me.default_account;
+        let wallets = default_account.wallets;
+
+        let btc_wallet = wallets
+            .clone()
+            .into_iter()
+            .find(|wallet| wallet.wallet_currency == WalletCurrency::BTC);
+
+        let usd_wallet = wallets
+            .into_iter()
+            .find(|wallet| wallet.wallet_currency == WalletCurrency::USD);
```

Comment:
I think for this you could use [partition_map](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.partition_map) from the itertools crate. Then you don't need to `clone`.